Library/Homebrew/cask/dsl.rb

```diff
@@ -243,6 +243,24 @@ def sha256(arg = nil)
       end
     end
 
+    # @api public
+    def arch(arm: nil, intel: nil)
+      should_return = arm.blank? && intel.blank?
+
+      set_unique_stanza(:arch, should_return) do
+        @on_system_blocks_exist = true
+
+        # If arm and intel are arrays and one isn't specified, default to an array that is the same size as the other
+        empty_value = Array.new(arm&.count || intel&.count) if arm.is_a?(Array) || intel.is_a?(Array)
```

Comment:
Sorry, my apologies for poor communication: I'm proposing that this only accepts values and not arrays and that they array form we move into two separate declarations e.g.

```ruby
arch arm: "arm", intel: "intel"
livecheck_arch = something_else arm: "larm", intel: "lintel"
```

Reply:
Hmm, this is what I thought you meant by

> My opinion is probably "less intuitive system" for those 👍🏻

What is the "less intuitive system" in your mind?

Reply:
The example I've had here is what I considered the "less intuitive" system although it's probably "more intuitive" 😅

Reply:
Hahaha okay, gotcha! I'm okay with either one since it's so few cases. Let me revert this back to include the original DSL proposal and see if that's what you're looking for.